packages/core/src/test/shared/clients/ec2Client.test.ts

```diff
@@ -77,51 +64,45 @@ const incompleteReservationsList: EC2.ReservationList = [
     },
 ]
 
-const incomepleteInstanceList: EC2.InstanceList = [
+const incomepleteInstanceList: Instance[] = [
     { InstanceId: 'running-1' },
     { InstanceId: 'stopped-2', Tags: [] },
     { InstanceId: 'pending-3', Tags: [{ Key: 'Name', Value: 'name3' }] },
 ]
 
+const getStatus: (i: string) => Promise<InstanceStateName> = (i) =>
+    new Promise((resolve) => {
+        resolve(i.split('-')[0] as InstanceStateName)
+    })
+
 describe('extractInstancesFromReservations', function () {
     const client = new Ec2Client('')
 
-    it('returns empty when given empty collection', async function () {
-        const actualResult = await client
-            .getInstancesFromReservations(
-                toCollection(async function* () {
-                    yield []
-                }) as AsyncCollection<EC2.ReservationList>
-            )
-            .promise()
+    it('returns empty when given empty collection', function () {
+        const actualResult = client.getInstancesFromReservations([])
 
         assert.strictEqual(0, actualResult.length)
     })
 
-    it('flattens the reservationList', async function () {
-        const actualResult = await client
-            .getInstancesFromReservations(intoCollection([completeReservationsList]))
-            .promise()
+    it('flattens the reservationList', function () {
+        const actualResult = client.getInstancesFromReservations(completeReservationsList)
         assert.deepStrictEqual(actualResult, completeInstanceList)
     })
 
     it('handles undefined and missing pieces in the ReservationList.', async function () {
-        const actualResult = await client
-            .getInstancesFromReservations(intoCollection([incompleteReservationsList]))
-            .promise()
+        const actualResult = client.getInstancesFromReservations(incompleteReservationsList)
         assert.deepStrictEqual(actualResult, incomepleteInstanceList)
     })
 })
 
 describe('updateInstancesDetail', async function () {
-    let client: MockEc2Client
-
+    let client: Ec2Client
     before(function () {
-        client = new MockEc2Client('test-region')
+        client = new Ec2Client('test-region')
```

Comment:
👀 does/will the new client help avoid mocks in some cases?

Reply:
For SDKv3, the SDK team appears to recommend this package for mocking the clients: https://aws.amazon.com/blogs/developer/mocking-modular-aws-sdk-for-javascript-v3-in-unit-tests/. 

With this light refactor I also removed the hand-crafted mocks when possible.